packages/bridge-ui-v2/src/libs/bridge/ETHBridge.ts

```diff
@@ -109,12 +109,19 @@ export class ETHBridge extends Bridge {
     const { messageStatus, destBridgeContract } = await super.beforeClaiming(args);
 
     let txHash: Hash;
-    const { msgHash, message } = args;
+    const { msgHash, message, receipt } = args;
     const srcChainId = Number(message.srcChainId);
     const destChainId = Number(message.destChainId);
 
     if (messageStatus === MessageStatus.NEW) {
-      const proof = await this._prover.encodedSignalProof(msgHash, srcChainId, destChainId);
+      const hops = routingContractsMap[srcChainId][destChainId].hops;
+
+      let proof;
+      if (hops.length > 0) {
```

Comment:
If hops are optional this will fail, just a heads up :)

Reply:
Update.